src/main/java/teammates/ui/webapi/EnrollStudentsAction.java

```diff
@@ -102,6 +102,7 @@ public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOpera
                         Student newStudent = new Student(
                                 course, enrollRequest.getName(),
                                 enrollRequest.getEmail(), enrollRequest.getComments(), team);
+                        newStudent.setId(sqlLogic.getStudentForEmail(courseId, enrollRequest.getEmail()).getId());
```

Comment:
@NicolasCwy If I'm not wrong, they are setting the ID so that they can get it back again within `updateStudentCascade`. To be fair, I was also initially confused about this when I last reviewed.

@marquestye I think in this case, if it's confusing, maybe we can leave a small comment explaining why this is done.

Reply:
I see so this block is if the the student was re-enrolled for the course, then instead of generating a new student, we would be updating the old student?